### ISSUE TITLE
Error: Failed finding DRM devices

### ISSUE THREAD

**Issue by tripzero** [ulissesf/qmassa#4]
When running with sudo, qmassa exists with:
```bash
Error: Failed finding DRM devices

Caused by:
    No such file or directory (os error 2)
```

strace reveals it is trying to open /sys/devices/power/events/energy-gpu.unit which does not exist on this device.  Am I missing a kernel config option?  Device is Asus Zenbook S Lunar Lake.

**Comment by lucasdemarchi** [comment]
more likely a patch missing. In Ubuntu 24.10, are you using 6.11? The config is CONFIG_PERF_EVENTS_INTEL_RAPL

**Comment by lucasdemarchi** [comment]
LNL is supported since commit acf68d98cae8 ("perf/x86/rapl: Add support for Intel Lunar Lake"), which is there since v6.10.

**Comment by ulissesf** [comment]
Thanks for the report, @tripzero.

Thanks, @lucasdemarchi. Yeah, weird it was not there through perf, might be missing a kernel config in Canonical's default config. @tripzero , can you try to check if that config option is missing? I'll just disable power if the events are not there (later tonight) as I wanted to avoid having to read through MSR, is possible.

**Comment by tripzero** [comment]
I'm using DRM tip with ubuntu's config.  I have CONFIG_PERF_EVENTS_INTEL_RAPL=m set.  What module does this option produce?  Perhaps it's just not loading?

**Comment by lucasdemarchi** [comment]
Yes. It seems that when the module is loaded  it reads the value and gets 0 so it doesn't enable it.  After patching it I can successfully read the gpu power consumption via perf.  I'll double check the proper way to fix it and submit a patch to the kernel

**Comment by lucasdemarchi** [comment]
@tripzero  the module is actually rapl. After hacking it I get:

```
LNL:~$ sudo ls -l /sys/devices/power/events/
...
-r--r--r-- 1 root root 4096 Nov  8 21:44 energy-gpu
-r--r--r-- 1 root root 4096 Nov  8 21:44 energy-gpu.scale
-r--r--r-- 1 root root 4096 Nov  8 21:44 energy-gpu.unit
...
LNL:~$ sudo modprobe -r rapl
LNL:~$ sudo ls -l /sys/devices/power/events/
ls: cannot access '/sys/devices/power/events/': No such file or directory
```

**Comment by ulissesf** [comment]
@tripzero I'll merge a patch to proceed without the power reporting until @lucasdemarchi merges his kernel patch for perf or when I add getting the power from MSR directly.

**Comment by ulissesf** [comment]
@tripzero it seems kernel will take some more time to integrate patch for always creating the energy-gpu perf event so I added fallback code to read directly from MSR.
